flink-runtime/src/test/java/org/apache/flink/runtime/scheduler/benchmark/e2e/RestartStreamingJobBenchmark.java

```diff
@@ -0,0 +1,96 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License
+ */
+
+package org.apache.flink.runtime.scheduler.benchmark.e2e;
+
+import org.apache.flink.runtime.clusterframework.types.ResourceProfile;
+import org.apache.flink.runtime.concurrent.ComponentMainThreadExecutorServiceAdapter;
+import org.apache.flink.runtime.execution.ExecutionState;
+import org.apache.flink.runtime.executiongraph.ExecutionVertex;
+import org.apache.flink.runtime.jobgraph.JobVertex;
+import org.apache.flink.runtime.jobmaster.slotpool.SlotPoolUtils;
+import org.apache.flink.runtime.scheduler.DefaultScheduler;
+
+import java.util.Collections;
+import java.util.concurrent.CompletableFuture;
+import java.util.stream.StreamSupport;
+
+/**
+ * The benchmark of restarting tasks in a STREAMING/BATCH job. The related method is {@link
+ * DefaultScheduler#handleGlobalFailure}.
+ */
+public class RestartStreamingJobBenchmark extends SchedulingAndDeployingBenchmark {
```

Comment:
`RestartStreamingJobBenchmark` ->  I prefer `HandleGlobalFailureAndRestartAllTasksBenchmark` here, and this is not just for `streaming` jobs.